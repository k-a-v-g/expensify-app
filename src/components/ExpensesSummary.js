import React from "react";
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from "numeral";

export const ExpensesSummary = (props) => {
    const expenseCount = props.expenses.length;
    const expensesTotal = selectExpensesTotal(props.expenses);
    return (
        <div>
            {props.expenses.length === 1 ? (
                <p>Viewing 1 expense totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
            ) : (
                <p>Viewing {expenseCount} expenses totalling {numeral(expensesTotal / 100).format('$0,0.00')}</p>
            )}
        </div>
    )
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);