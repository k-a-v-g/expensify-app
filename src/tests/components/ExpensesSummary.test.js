import { ExpensesSummary } from "../../components/ExpensesSummary";
import React from "react";
import { shallow } from "enzyme";
import expenses from '../fixtures/expenses';

test('should render details of one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render details of multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render summary correctly when no expenses are displayed', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]} /> );
    expect(wrapper).toMatchSnapshot();
});