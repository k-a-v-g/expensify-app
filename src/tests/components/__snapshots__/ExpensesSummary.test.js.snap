// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render details of multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render details of one expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling 
    $1,095.00
  </p>
</div>
`;

exports[`should render summary correctly when no expenses are displayed 1`] = `
<div>
  <p>
    Viewing 
    0
     expenses totalling 
    $0.00
  </p>
</div>
`;
